engine/runtime/src/test/java/org/enso/interpreter/test/MetaObjectTest.java

```diff
@@ -215,6 +217,25 @@ public void typesOfConstructors() throws Exception {
     }
   }
 
+  @Test
+  public void nothingIsNotMeta() {
+    var g = ValuesGenerator.create(ctx, ValuesGenerator.Language.ENSO);
+    var nothing = g.typeNothing();
+    assertThat("Nothing is not meta", nothing.isMetaObject(), is(false));
+  }
+
+  @Test
+  public void nothingWithWarningIsNotMeta() {
+    var src = """
+import Standard.Base.Warning.Warning
+import Standard.Base.Nothing.Nothing
+
+main = Warning.attach "foo" Nothing
+""";
+    var nothingWithWarn = TestBase.evalModule(ctx, src);
+    assertThat(nothingWithWarn.isMetaObject(), is(false));
```

Comment:
@JaroslavTulach This test case fails with
```
[error] Test org.enso.interpreter.test.MetaObjectTest.nothingWithWarningIsNotMeta failed: org.graalvm.polyglot.PolyglotException: java.lang.AssertionError: Invariant contract violation for receiver Nothing(org.enso.interpreter.runtime.data.Type)., took 3.592 sec
[error]     at com.oracle.truffle.api.interop.InteropLibrary$Asserts.getMetaParents(InteropLibrary.java:5209)
[error]     at com.oracle.truffle.api.interop.InteropLibraryGen.genericDispatch(InteropLibraryGen.java:498)
[error]     at com.oracle.truffle.api.library.ReflectionLibraryDefault$Send.doSendCached(ReflectionLibrary.java:182)
[error]     at com.oracle.truffle.api.library.ReflectionLibraryDefaultGen$ReflectionLibraryExports$Cached.executeAndSpecialize(ReflectionLibraryDefaultGen.java:140)
[error]     at com.oracle.truffle.api.library.ReflectionLibraryDefaultGen$ReflectionLibraryExports$Cached.send(ReflectionLibraryDefaultGen.java:106)
[error]     at com.oracle.truffle.api.library.ReflectionLibraryGen$CachedDispatch.send(ReflectionLibraryGen.java:283)
[error]     at org.enso.interpreter.runtime.error.WithWarnings.send(WithWarnings.java:168)
[error]     at org.enso.interpreter.runtime.error.WithWarningsGen$ReflectionLibraryExports$Cached.send(WithWarningsGen.java:402)
[error]     at com.oracle.truffle.api.interop.InteropLibraryGen$Proxy.getMetaParents(InteropLibraryGen.java:3808)
[error]     at com.oracle.truffle.api.interop.InteropLibrary$Asserts.assertNoMetaObject(InteropLibrary.java:5094)
[error]     at com.oracle.truffle.api.interop.InteropLibrary$Asserts.isMetaObject(InteropLibrary.java:5070)
[error]     at com.oracle.truffle.polyglot.PolyglotValueDispatch$InteropValue$IsMetaObjectNode.doCached(PolyglotValueDispatch.java:4854)
[error]     at com.oracle.truffle.polyglot.PolyglotValueDispatchFactory$InteropValueFactory$IsMetaObjectNodeGen.executeAndSpecialize(PolyglotValueDispatchFactory.java:11021)
[error]     at com.oracle.truffle.polyglot.PolyglotValueDispatchFactory$InteropValueFactory$IsMetaObjectNodeGen.executeImpl(PolyglotValueDispatchFactory.java:10969)
[error]     at com.oracle.truffle.polyglot.HostToGuestRootNode.execute(HostToGuestRootNode.java:124)
[error]     at org.graalvm.polyglot.Value.isMetaObject(Value.java:219)
[error]     at org.enso.interpreter.test.MetaObjectTest.nothingWithWarningIsNotMeta(MetaObjectTest.java:237)
[error]     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
[error]     at java.lang.reflect.Method.invoke(Method.java:580)
[error]     ...
```

(Run it with `sbt:runtime> testOnly *MetaObjectTest -- -z org.enso.interpreter.test.MetaObjectTest.nothingWithWarningIsNotMeta`).

The test case above succeeds. Shouldn't we fix this as well?

Reply:
Note that this test case is a minimal reproducer of https://github.com/enso-org/enso/pull/8595/files/818c1ef2910c9f1e5608b2aee4da162a604d2fe1..8b685a136549fb11809ab4d8f23f171192688ba3. I have debugged the internal Truffle code and saw that for some reason, it calls `isMetaObject` on a Nothing with a warning attached.

Reply:
cc @hubertp

Reply:
Answer to myself: Yes, that should not throw `AssertionError`. The `isMetaObject` on that object should return just false.

Reply:
We certainly shouldn't be getting:

> java.lang.AssertionError: Invariant contract violation for receiver Nothing

but rather `false` or no answer, etc.